packages/compiler-sfc/src/templateUtils.ts

```diff
@@ -2,15 +2,13 @@ import { UrlWithStringQuery, parse as uriParse } from 'url'
 import { isString } from '@vue/shared'
 
 export function isRelativeUrl(url: string): boolean {
-  const firstChar = url.charAt(0)
-  return firstChar === '.' || firstChar === '~' || firstChar === '@'
+  return url.startsWith('.') || url.startsWith('~') || url.startsWith('@')
```

Comment:
Another approach would be:-
```js
return ['.', '~', '@'].some(char => url.startsWith(char));
```
But, this is expensive.

Reply:
Not reasonable. This could lead 3 calls. The original just 1 call.

Reply:
Also applies to the proposed `.startsWith()` implementation.

Using a single `.charAt()` call is probably simply preferable here, or directly accessing it through `url[0]`